pages/thisWeek.js

```diff
@@ -0,0 +1,44 @@
+import WeeklyCards from "@/components/WeeklyCards";
+import Link from "next/link";
+
+export default function ThisWeek({ events }) {
+  //prevent the hydration error on refresh
+  if (!events || events.length === 0) {
+    return;
+  }
+
+  function getEachDaysEvents() {
+    const weekdayEvents = {};
+    weekdayEvents.monday = events.filter((event) => {
+      return event.weekday === "Monday";
+    });
+    weekdayEvents.tuesday = events.filter((event) => {
+      return event.weekday === "Tuesday";
+    });
+    weekdayEvents.wednesday = events.filter((event) => {
+      return event.weekday === "Wednesday";
+    });
+    weekdayEvents.thursday = events.filter((event) => {
+      return event.weekday === "Thursday";
+    });
+    weekdayEvents.friday = events.filter((event) => {
+      return event.weekday === "Friday";
+    });
+
+    return weekdayEvents;
+  }
```

Comment:
I'm a little confused as to what's happening here. 🤯

Reply:
this does not need to be a function you can do this just like this, directly in your component function

```js
  const weekdayEvents = {
    monday: events.filter((event) => {
      return event.weekday === "Monday";
    }),
    tuesday: events.filter((event) => {
      return event.weekday === "Tuesday";
    }),
   wednesday: events.filter((event) => {
      return event.weekday === "Wednesday";
    }),
   thursday: events.filter((event) => {
      return event.weekday === "Thursday";
    }),
   friday: events.filter((event) => {
      return event.weekday === "Friday";
    })
  };
```